gprofiler/cloud_metadata.py

```diff
@@ -0,0 +1,161 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Union
+
+import requests
+from requests import Response
+
+from gprofiler.exceptions import BadResponseCode
+from gprofiler.log import get_logger_adapter
+
+METADATA_REQUEST_TIMEOUT = 5
+
+logger = get_logger_adapter(__name__)
+
+
+@dataclass
+class InstanceMetadataBase:
+    provider: str
+
+
+@dataclass()
+class AwsInstanceMetadata(InstanceMetadataBase):
+    region: str
+    zone: str
+    instance_type: str
+    life_cycle: str
+    account_id: str
+    image_id: str
+    instance_id: str
+
+
+@dataclass
+class GcpInstanceMetadata(InstanceMetadataBase):
+    zone: str
+    instance_type: str
+    preempted: bool
+    preemptible: bool
+    instance_id: str
+    image_id: str
+    name: str
+
+
+@dataclass
+class AzureInstanceMetadata(InstanceMetadataBase):
+    instance_type: str
+    zone: str
+    region: str
+    subscription_id: str
+    resource_group_name: str
+    resource_id: str
+    instance_id: str
+    name: str
+    image_info: Optional[Dict[str, str]]
+
+
+def get_aws_metadata() -> Optional[AwsInstanceMetadata]:
+    metadata_response = send_request("http://169.254.169.254/latest/dynamic/instance-identity/document")
+    life_cycle_response = send_request("http://169.254.169.254/latest/meta-data/instance-life-cycle")
+    if life_cycle_response is None or metadata_response is None:
+        return None
+    instance_metadata = metadata_response.json()
+    region = instance_metadata["region"]
+    zone = instance_metadata["availabilityZone"]
+    instance_type = instance_metadata["instanceType"]
+    account_id = instance_metadata["accountId"]
+    image_id = instance_metadata["imageId"]
+    instance_id = instance_metadata["instanceId"]
+    life_cycle = life_cycle_response.text
+    return AwsInstanceMetadata("aws", region, zone, instance_type, life_cycle, account_id, image_id, instance_id)
```

Comment:
Can you make it shorter somehow? For example, you can use the exact field names of AWS (also makes it more readable), then you can just `return AwsInstance(**instance_metadata)`

Reply:
It won't work on all of the cloud providers anyway, and also somewhat changed in #134

Reply:
k